vllm/distributed/kv_transfer/kv_connector/v1/nixl_connector.py

```diff
@@ -256,10 +256,7 @@ def get_num_new_matched_tokens(
 
         if params is not None and params.get("do_remote_prefill"):
             # Remote prefill: get all prompt blocks from remote.
-            assert num_computed_tokens % self.block_size == 0
-            rounded_num_prompt_tokens = round_down(
-                len(request.prompt_token_ids), self.block_size)
-            count = max(rounded_num_prompt_tokens - num_computed_tokens, 0)
+            count = max(len(request.prompt_token_ids) - num_computed_tokens, 0)
```

Comment:
No need for the `max` now
```suggestion
            count = len(request.prompt_token_ids) - num_computed_tokens
```